browser/brave_ads/ads_tab_helper.cc

```diff
@@ -148,10 +149,12 @@ void AdsTabHelper::DocumentOnLoadCompletedInMainFrame() {
       std::make_unique<dom_distiller::SourcePageHandleWebContents>(
           web_contents(), false);
 
-  content::RenderFrameHost* render_frame_host =
+  // We can't use |render_frame_host| directly because it represents the frame
```

Comment:
not sure I understand this. The method is `DocumentOnLoadCompletedInMainFrame`, but it gets called in all frames? If that's the case we need to exit early if this is not the main frame or this will run once for every frame.

Reply:
I think you're right. I was being extra paranoid here for no reason, but now I look again into it I think I got confused by the fact that in the renderer process you can actually have a main frame that is **not** a root frame (i.e. the main local frame for an iframe in the process handled that iframe), but that can't happen here being the browser process.

Fixed in https://github.com/brave/brave-core/commit/4f7128ffc053799d403998ea6cd64aad7fc55d79